src/main/java/com/datadoghq/sketch/ddsketch/store/DenseStore.java

```diff
@@ -47,7 +47,7 @@
     DenseStore(DenseStore store) {
         this.arrayLengthGrowthIncrement = store.arrayLengthGrowthIncrement;
         this.arrayLengthOverhead = store.arrayLengthOverhead;
-        this.counts = store.counts == null ? null : Arrays.copyOf(store.counts, store.counts.length);
+        this.counts = store.counts == null || store.isEmpty() ? null : Arrays.copyOf(store.counts, store.counts.length);
```

Comment:
I believe that even when it's not empty, we should copy only the used part of the array rather than the full array (or slightly more, as per the array growing logic, that is, the size of `this.counts` should be `getNewLength(store.minIndex, store.maxIndex)`, and non-empty counts should be centered).

Reply:
Sure, I can make that improvement.

Reply:
I did this [here](https://github.com/DataDog/sketches-java/pull/34/commits/f52812390fc2ee0c707aed8a5b2ac246f5ba1653)